languagetool-language-modules/ar/src/test/java/org/languagetool/rules/ar/ArabicMasdarToVerbFilterTest.java

```diff
@@ -0,0 +1,109 @@
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301
+ * USA
+ */
+package org.languagetool.rules.ar;
+
+import org.junit.Test;
+import org.languagetool.AnalyzedTokenReadings;
+import org.languagetool.rules.FakeRule;
+import org.languagetool.rules.RuleMatch;
+import org.languagetool.rules.SimpleReplaceDataLoader;
+import org.languagetool.rules.patterns.RuleFilter;
+import org.languagetool.tagging.ar.ArabicTagger;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static java.util.Arrays.asList;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.IsNull.notNullValue;
+
+public class ArabicMasdarToVerbFilterTest {
+
+  private final RuleMatch match = new RuleMatch(new FakeRule(), null, 0, 10, "message");
+  private final RuleFilter filter = new ArabicWordinessRule.MasdarToVerbFilter();
+  private final ArabicTagger tagger = new ArabicTagger();
+  private static final String FILE_NAME = "/ar/arabic_masdar_verb.txt";
+  //  boolean debug = true;
+  final boolean debug = false;
+
+  @Test
+  public void testFilter() throws IOException {
+    assertSuggestion("عمل", "يعمل", false);
+    assertSuggestion("إعمال", "يعمل", false);
+    assertSuggestion("سؤال", "يسأل", false);
+    assertSuggestion("أكل", "يأكل", false);
+  }
+
+  private void assertSuggestion(String word, String expectedSuggestion, boolean debug) throws IOException {
+
+    String word1 = "يقوم";
+    String word2 = "بال" + word;
+    Map<String, String> args = new HashMap<>();
+    args.put("verb", word1);
+    args.put("noun", word2);
+    List<AnalyzedTokenReadings> patternTokens = tagger.tag(asList(word1, word2));
+    AnalyzedTokenReadings[] patternTokensArray = patternTokens.stream().toArray(AnalyzedTokenReadings[]::new);
+    RuleMatch ruleMatch = filter.acceptRuleMatch(match, args, -1, patternTokensArray);
+    if (!debug) {
+      assertThat(ruleMatch.getSuggestedReplacements().size(), is(1));
+      assertThat(ruleMatch.getSuggestedReplacements().get(0), is(expectedSuggestion));
+    } else { //  debug is true
+      String suggestion = "";
+      if (!ruleMatch.getSuggestedReplacements().isEmpty()) {
+        suggestion = ruleMatch.getSuggestedReplacements().toString();
+      }
+      // show only no suggestion cases
+      if (!suggestion.startsWith("[ي")) {
+        System.out.println("مثال: " + word1 + " " + word2 + " مقترح:" + suggestion);
```

Comment:
There are usually not `System.out.println` in tests other than those that print progress. Is this an error? Then the test should fail.